### PR TITLE
Close generate files

### PR DESCRIPTION
**Pull request by externl**
Previously, several of our slice compilers had `closeOutput` functions which were unused. I've removed this function and moved the logic to the destructors.

### CODE REVIEW

File: cpp/src/slice2cpp/Gen.cpp [hunk 1]
```diff
@@ -635,6 +635,11 @@ Slice::Gen::~Gen()
     H << "\n\n#include <IceUtil/PopDisableWarnings.h>";
     H << "\n#endif\n";
     C << '\n';
+
+    H.close();
+    C.close();
+    implH.close();
```
Comment (pepone) [inline cpp/src/slice2cpp/Gen.cpp:641]: I thought we already removed `--impl`, is this a leftover?

File: cpp/src/slice2js/Gen.cpp [hunk 2]
```diff
@@ -673,11 +673,13 @@ Slice::Gen::~Gen()
     if (_jsout.isOpen() || _useStdout)
     {
         _jsout << '\n';
+        _jsout.close();
```
Comment (pepone) [inline cpp/src/slice2js/Gen.cpp:676]: do we need to call `close` when `_useStdout` is true? previously we didn't.

**Review by pepone — APPROVED** [review]

File: cpp/src/slice2js/Gen.cpp (hunk 2 above)
Comment (externl) [reply cpp/src/slice2js/Gen.cpp:676]: We don't need to, but we don't really need to for the files either. All the handles get closed when the process exits. It's not an error to do it, so I think it's fine here.

File: cpp/src/slice2cpp/Gen.cpp (hunk 1 above)
Comment (externl) [reply cpp/src/slice2cpp/Gen.cpp:641]: Could be! Maybe I can remove it.

Comment (externl) [reply cpp/src/slice2cpp/Gen.cpp]: Removed!

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**